Yosemite/Yosemite/Actions/CouponAction.swift

```diff
@@ -27,4 +27,12 @@ public enum CouponAction: Action {
     case deleteCoupon(siteID: Int64,
                       couponID: Int64,
                       onCompletion: (Result<Void, Error>) -> Void)
+
+    /// Updates a coupon for a site given its ID
+    ///
+    /// - `coupon`: the coupon to be updated.
+    /// - `onCompletion`: invoked when the deletion finishes.
+    ///
+    case updateCoupon(_ coupon: Coupon,
+                      onCompletion: (Result<Void, Error>) -> Void)
```

Comment:
nit: do you think it makes sense to return the coupon in the result, or does the app only use the coupons from the storage layer? In case the update on some properties could be rejected in the backend in the update request

Reply:
Interesting 👍 Are you suggesting checking the properties and showing some error if something doesn't match?

Reply:
I think just returning the whole coupon from the API response is good, like the suggested diff. Then it's up to the caller whether to use the coupon from the API update

```suggestion
                      onCompletion: (Result<Coupon, Error>) -> Void)
```

Reply:
I figure this should be necessary to keep track of the current state of a coupon in its update form, so I added the coupon in the result in 76f69c126b5b7c0a292e4d2e56dad0d830000110. Thanks for the suggestion!